target/linux/ath79/image/gen_kt_img.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+SUFFIX=.patched
+TMP_FILE1="/tmp/__tmp_file1__"
+TMP_FILE2="/tmp/__tmp_file2__"
+TMP_FILE3="/tmp/__tmp_file3__"
+
+function usage() {
+  echo ""
+  echo "$0 (bin file)"
+  echo ""
+}
+
+function patch_uimage_header() {
+  if [ ! -f ${1} ]
+  then
+    echo "File not found !"
+    exit 1
+  fi
+
+  dd if=${1} of=${TMP_FILE1} count=64 bs=1
+
+  dd if=${TMP_FILE1} of=${TMP_FILE2} count=4 bs=1
+  printf '\x00\x00\x00\x00\x00\x02\x04\x00' | dd of=${TMP_FILE2} bs=1 seek=4 conv=notrunc
+  dd bs=1 skip=8 if=${TMP_FILE1} >> ${TMP_FILE2}
+
+  CRC32_CHECKSUM=$(crc32 ${TMP_FILE2})
+  echo "*** CRC32 Checksum: ${CRC32_CHECKSUM}"
+
+  echo "${CRC32_CHECKSUM}" | xxd -r -p > ${TMP_FILE3}
+  dd if=${TMP_FILE3} of=${TMP_FILE2} bs=1 seek=4 conv=notrunc
+  
+  dd if=${TMP_FILE2} of=${1}$SUFFIX
+  
+  dd bs=64 skip=1 if=${1} of=${1}$SUFFIX oflag=append conv=notrunc
+  
+  yes | cp -f ${1}$SUFFIX ${1}
```

Comment:
Isn't the point of -f that you don't need input?

Apart from that, shellcheck tells me the pipe isn't working anyway.